client/scss/tools/_various.colors.scss

```diff
@@ -0,0 +1,62 @@
+// $color is either a color or an hsl tuple
+@mixin define-color($name, $color) {
+    $h: null;
+    $s: null;
+    $l: null;
+
+    @if type-of($color) == color {
+        $h: hue($color) / 1deg; // Cast to unitless
+        $s: saturation($color);
+        $l: lightness($color);
+    } @else {
+        $h: nth($color, 1);
+        $s: nth($color, 2);
+        $l: nth($color, 3);
+    }
+
+    --#{$name}-hue: #{$h};
+    --#{$name}-saturation: #{$s};
+    --#{$name}-lightness: #{$l};
+    --#{$name}: hsl(#{ var(--#{$name}-hue), var(--#{$name}-saturation), var(--#{$name}-lightness) });
+}
+
+@function get-color($name) {
+    @return (var(--#{$name}-hue), var(--#{$name}-saturation), var(--#{$name}-lightness));
+}
+
+@function css-darken($hsl-tuple, $darken-by) {
+    $h: nth($hsl-tuple, 1);
+    $s: nth($hsl-tuple, 2);
+    $l: nth($hsl-tuple, 3);
+    @return ($h, $s, calc(#{$l} - #{$darken-by + 0%}));
```

Comment:
Is this and the other HSL shifts really equivalent to the Sass color functions they replace?

Reply:
Near as I can tell, yep! Run the following through sass (I did it on sassmeister.com to check easily) and you'll see the two colors end up identical:
```scss
.test {
    color: darken(red, 10%);
    background-color: hsl(hue(red), saturation(red), lightness(red) - 10%);
}
```